ast35/Python/ast.c

```diff
@@ -1247,6 +1247,22 @@ ast_for_arg(struct compiling *c, const node *n)
     return ret;
 }
 
+static int
+set_arg_comment_annotation(struct compiling *c, arg_ty arg, node *tc)
+{
+    if (arg->annotation) {
+        ast_error(c, tc,
+                "annotated arg has associated type comment");
```

Comment:
Also worth clarifying in the PEP that if you annotate an argument in PEP 3107 fashion, you can't also give it a type comment -- people should be able to guess this one, but it'd be better to be unambiguous.